hazelcast/src/main/java/com/hazelcast/util/HashUtil.java

```diff
@@ -279,29 +292,38 @@ public static long MurmurHash3_x64_64(final byte[] data, int offset, int len, fi
         h1 = MurmurHash3_fmix(h1);
         h2 = MurmurHash3_fmix(h2);
 
-        h1 += h2;
-        //h2 += h1; // unused, used only for 128-bit version
-        return h1;
+        return h1 + h2;
     }
 
-    public static long MurmurHash3_x64_64_direct(long address, int offset, int len) {
-        return MurmurHash3_x64_64_direct(address, offset, len, DEFAULT_MURMUR_SEED);
+    public static long MurmurHash3_x64_64(byte[] data, int offset, int len) {
+        return MurmurHash3_x64_64(data, offset, len, DEFAULT_MURMUR_SEED);
     }
 
-    public static long MurmurHash3_x64_64_direct(long address, int offset, int len, final int seed) {
+
+    public static long MurmurHash3_x64_64_direct(MemoryAccessor mem, long base, int offset, int len, final int seed) {
+
+        final boolean memIsLittleEndian = !mem.isBigEndian();
+
+        // (len & ~(MURMUR64_BLOCK_SIZE - 1)) is the length rounded down to the Murmur64 block boundary
+        final long tailStart = base + offset + (len & ~(MURMUR64_BLOCK_SIZE - 1));
+
         long h1 = 0x9368e53c2f6af274L ^ seed;
         long h2 = 0x586dcd208f7cd3fdL ^ seed;
 
         long c1 = 0x87c37b91114253d5L;
         long c2 = 0x4cf5ad432745937fL;
 
-        long k1 = 0;
-        long k2 = 0;
-
-        for (int i = 0; i < len / 16; i++) {
-            k1 = MurmurHash3_getBlock_direct(address, (i * 2 * 8) + offset);
-            k2 = MurmurHash3_getBlock_direct(address, ((i * 2 + 1) * 8) + offset);
-
+        long k1;
+        long k2;
+
+        for (long blockAddr = base + offset; blockAddr < tailStart; blockAddr += MURMUR64_BLOCK_SIZE) {
+            if (memIsLittleEndian) {
+                k1 = mem.getLong(blockAddr);
```

Comment:
same comment about unaligned memory access if the given `MemoryAccessor` is not alignment aware and `offset` is not 4 bytes aligned